### PR TITLE
Make text bigger each time it stacks

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Chat entries that get stacked will have their sizes increase gradually based on their stack count.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs [hunk 1]
```diff
@@ -172,20 +173,34 @@ public void SetText(string message, TMP_SpriteAsset languageSprite, TMP_FontAsse
...
 			stackObject.SetActive(true);
 			AnimateFade(1f, 0f);
 			StartCoroutine(AnimateStackObject());
+			StackMessageSizeIncrease();
 			this.RestartCoroutine(FadeCooldown(), ref fadeCooldownCoroutine);
 		}
 
+		private void StackMessageSizeIncrease()
+		{
+			string pattern = @"<size(=|\+=|\+)([0-9]+|[0-9]+\+|=[0-9]+\+)>";
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: what's with the + at the end? Can the number be something like ``10+``?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: yes, it can be like this:

<size=[number]+>
<size=+[number]>
<size+=[number]>

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: what are all the possible cases? Could it ever be ``<size+10`` for example?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: I just posted all the possible cases

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: image

your second case doesn't match and it matches a case that shouldn't be valid

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: what characters do you want to capture from the matches? I'm guessing the symbol and the value, right?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: image
this one should be enough

``<size(=|\+=|=+)(\+?[0-9]+\+?)>``

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: I pushed a commit that should resolve this:
`<Size(=|\+?=)?([0-9]+|\+[0-9]+|[0-9]+\+)?>`

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: ``<size(=|\+=)(\+?[0-9]+\+?)>``

I had one unused alternative inside the first group

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: nevermind, it broke completely ingame

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: > `<size(=|\+=)(\+?[0-9]+\+?)>`
> 
> I had one unused alternative inside the first case

image

this works, but now running into this funny issue

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: that's funny. Could it be because of the multiline? Do you have that flag?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: removed it, but problem still presists

File: UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs [hunk 2]
```diff
@@ -172,20 +173,35 @@ public void SetText(string message, TMP_SpriteAsset languageSprite, TMP_FontAsse
...
+		private void StackMessageSizeIncrease()
+		{
+			string pattern = @"<size(=|\+=)(\+?[0-9]+\+?)>";
+			Match match = Regex.Match(messageText.text, pattern);
+			if (match.Success)
+			{
+				messageText.text = Regex.Replace(messageText.text, pattern, match =>
+				{
+					float number = float.Parse(match.Groups[2].Value);
+					Debug.Log(number);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Chat/ChatEntry.cs]: remove the debug log

**Review by corp-0 — APPROVED** [review]

**Pull request merged**